pyretailscience/utils/date.py

```diff
@@ -40,13 +40,21 @@ def filter_and_label_by_periods(
         ValueError: If any value in period_ranges is not a tuple of length 2.
     """
     branches = []
+    date_column = transactions[get_option("column.transaction_date")]
 
     for period_name, date_range in period_ranges.items():
         if not (isinstance(date_range, tuple) and len(date_range) == 2):  # noqa: PLR2004 - Explained in the error below
             msg = f"Period '{period_name}' must have a (start_date, end_date) tuple"
             raise ValueError(msg)
 
-        period_condition = transactions[get_option("column.transaction_date")].between(date_range[0], date_range[1])
+        start_date, end_date = date_range
+
+        if isinstance(start_date, datetime):
+            start_date = start_date.date()
+        if isinstance(end_date, datetime):
+            end_date = end_date.date()
+
+        period_condition = date_column.between(start_date, end_date)
```

Comment:
So I ran into another issue and I realized we need a fancier solution here. We need to look up at the type of the column and use ibis literal to get it to the right type. Something like

Outside the loop
```
date_col_dtype = transactions[date_column].type()
```

Inside the loop
```suggestion
        start_date, end_date = date_range
        period_condition = date_column.between(ibis.Literal(start_date, type=date_col_type), ibis.Literal(end_date, type=date_col_type)
```